@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -486,14 +470,29 @@ export class PoiRenderer {
      * @param env - TODO! The current zoomLevel level of {@link MapView}
      *
      * @returns `True` if the space is not already allocated by another object (text label or POI)
+     * or a Promise if the image has to be loaded first. This happens when the user explicitly
+     * doesn't load the image when it is put into the cache.
      */
-    prepareRender(pointLabel: TextElement, env: Env): boolean {
+    prepareRender(pointLabel: TextElement, env: Env): boolean | Promise<boolean> {
         const poiInfo = pointLabel.poiInfo;
         if (poiInfo === undefined) {
             return false;
         }
+        const { imageItem, imageTexture } = this.imageCached(poiInfo.imageTextureName);
+        if (!imageItem) {
+            // Set the POI to be invalid, so it will not be rendered, nor will it be re-checked in a
+            // future request. The item must exist (but not necessarily loaded in the cache before
+            // the POI can be initialized). We don't log an error, because the API of the
+            // MapViewImageCache allows the user to delete an image from the cache. The geometry
+            // creation is handled using some task queue which means that in some cases, a POI entry
+            // may be removed before the geometry is created. See queueGeometryCreation, which
+            // appends some item to the list. In such cases, the POI will just be abandoned and the
+            // geometry will be recreated with the new ImageItem.
+            poiInfo.isValid = false;
```

Comment:
In the Venues case, the image is not in cache cause the theme is not yet loaded. Here we'll mark pois referencing to images that are not yet loaded as invalid, and they'll never be rendered.

Do you plan to fix that on JSAPI side by waiting for the theme to load?

Reply:
I don't think it is needed, if I check the JSAPI side, the `setTheme`, then `await addDataSource`, which internally anyway `await`s on the theme.